floodviz/hydrograph_utils.py

```diff
@@ -19,10 +19,16 @@ def parse_hydrodata(jdata):
         corresponds to a different site.   
 
     """
-
     all_series_data = []
-
+    
+    if not isinstance(jdata, list):
+        return all_series_data
+    
     for idx, site in enumerate(jdata):
+        # Check that list elements are dicts
+        if not isinstance(site, dict):
+            return all_series_data
+
         site_name = site['sourceInfo']['siteName']
```

Comment:
This is one of those religious wars. Should there be a bunch of try except handlers or should you make an attempt to handle a dictionary whose format may not be what you expect? I think it's clearer to do the later and IMHO the code is cleaner. We've actually talked about this issue in my scrum team not to long ago.